encoder/vaapiencoder_base.cpp

```diff
@@ -341,6 +348,146 @@ static bool copyImage(uint8_t* destBase,
     return true;
 }
 
+static uint32_t get_bpp_in_1stplane_by_fourcc(uint32_t fourcc)
+{
+    uint32_t bpp;
+
+    switch (fourcc) {
+    case VA_FOURCC_NV12:
+    case VA_FOURCC_YV12:
+    case VA_FOURCC_I420:
+        bpp = 1;
+        break;
+
+    case VA_FOURCC_P010:
+    case VA_FOURCC_P016:
+        bpp = 2;
+        break;
+
+    default:
+        bpp = 1;
+        break;
+    }
+
+    return bpp;
+}
+
+static bool loadFrameToSurface(VAImage surface_image,uint8_t* dst,uint8_t* src,uint32_t fourcc)
+{
+    uint8_t *y_src, *u_src, *v_src;
+    uint8_t *y_dst, *u_dst, *v_dst;
+    void *surface_p = dst;
+    uint32_t i, row, col;
+
+    if (surface_image.format.fourcc == VA_FOURCC_YV12 ||
+        surface_image.format.fourcc == VA_FOURCC_I420 ||
+        surface_image.format.fourcc == VA_FOURCC_NV12 ||
+        surface_image.format.fourcc == VA_FOURCC_P010 ||
+        surface_image.format.fourcc == VA_FOURCC_P016) {
+        uint32_t bpp = get_bpp_in_1stplane_by_fourcc(surface_image.format.fourcc);
+
+        y_src = src;
+        if ((fourcc == VA_FOURCC('I', '0', '1', '0')) ||
+            (fourcc == VA_FOURCC_I420)) {
+            u_src = src + surface_image.width * surface_image.height * bpp;
+            v_src = src + surface_image.width * surface_image.height * 5 / 4 * bpp;
+        } else {
+            ERROR("only support I010/I420 ");
+            return false;
```

Comment:
We can return false if fourcc is not I010 or I420 at the first of this function to make code more clear.